### ISSUE TITLE
Support "Draft notes" bin for unfinished "New Note"s being taken.

### ISSUE THREAD

**Issue by Bujiraso** [orgzly-revived/orgzly-android-revived#220]
- [x] I have searched for existing issues that may be the same as or related to mine.

I regularly use the "new note" feature from long-pressing on Orgzly-Revived's icon or doing the same and dragging that to the home-screen. You can also select a default book for these features. This is great and I love it.

What I would love to see changed is how it handles navigation away from this view.
When I need to quickly switch apps for a notification or call, all of my writing is gone. I can't get back in. It's not in the recent app drawer, and hitting new note gives me a new note. There are a /few/ times I've managed to somehow resume, but I'm not even sure how.

# Proposed Solution
On navigation away from the "new note" page, submit it to a new "Drafts" area (could be found in the sidebar, or created as a new book, not important), and ensure that no data is lost while taking a note.

Please let me know if that idea seems foggy and I'm happy to clear it up anywhere it is, thanks!

**Comment by alensiljak** [comment]
A simple workaround is to save the note before switching.
For me, tapping on New Note shortcut returns to the editing screen.

**Comment by Bujiraso** [comment]
As mentioned on #434, this is almost exclusively concerning accidental switches away. I definitely save every time I can!

**Comment by alensiljak** [comment]
Yes, I lost a few this way, too. The best option would be to have the window shown in the Recent Apps, so that one can go back to it. And than any new one also stays active until saved.

Is this suggestion/issue specifically asking for the Drafts folder or is that only as a potential way of solving the issue of disappearing New Note contents? 
If so, I think we could close one of these issues.

**Comment by Bujiraso** [comment]
The drafts is a suggestion specifically asking to solve the disappearing new note, yes. It makes sense to me to close #434 since it's a dupe, but just due to age. I will edit the use case into the top message.

**Comment by amberin** [comment]
I believe the least surprising behavior here would be to persist the activity with its state intact, until it's explicitly killed. This is how most other apps behave (e.g. e-mail apps such as K9 or Outlook). If we had that, would we really need a "drafts" bin? In any case, that should be the first step in improving this.

**Comment by alensiljak** [comment]
Some things to check:

- `android:excludeFromRecents="true": If this flag is set in the activity's manifest, it prevents the activity from appearing in the recent apps list. Removing this flag resolves the issue.
- Launch Mode: Certain launchMode settings, like singleTask or singleTop, can affect how activities are displayed in recent apps. Testing with default launch modes may help.
- Shortcut Implementation: If the shortcut uses custom intents or URIs, they might bypass standard recent apps behavior

**Comment by Bujiraso** [comment]
> I believe the least surprising behavior here would be to persist the activity with its state intact, until it's explicitly killed. This is how most other apps behave (e.g. e-mail apps such as K9 or Outlook). If we had that, would we really need a "drafts" bin? In any case, that should be the first step in improving this.

As suggested but I'll state explicitly again, the Drafts idea was just my suggested solution. Ignore it entirely if there's a good way to make things not disappear! 😃

**Comment by alensiljak** [comment]
Do we know why the New Note window seems to disappear?

One positive thing here is that, if one clicks on the New Note shortcut (on home screen), the New Note will appear, together with all the content that was "lost". This is currently a workaround, I guess.

Another quick fix would be, on leaving the activity, to save the note in the destination notebook with "[Draft]" added at the beginning of the title. This would probably be a better option than creating a separate notebook. I use Inbox as the default destination for new notes and it would be ok to have the drafts automatically saved there.

**Comment by amberin** [comment]
> `android:excludeFromRecents="true": If this flag is set in the activity's manifest, it prevents the activity from appearing in the recent apps list. Removing this flag resolves the issue.

In any case, this is definitely step one, right? I suspect once we fix that, this won't be an issue at all.

**Comment by alensiljak** [comment]
ok, I reckon [this](https://github.com/orgzly-revived/orgzly-android-revived/blob/dee9cfbdf824e6796257ff7566479fb16193a824/app/src/main/AndroidManifest.xml#L68) is the place. The shortcut seems to send an Intent for ShareActivity, which then displays the NewNote fragment.
The effect will be that other fragments shown via Share activity will also be displayed in history but I guess that's a fair trade-off.

**Comment by Bujiraso** [comment]
> One positive thing here is that, if one clicks on the New Note shortcut again (i.e. on the home screen), the New Note will appear, together with all the content that was "lost". This is currently a workaround, I guess.

Hm, I don't see this on `1.10.0 (fdroid)`.

My steps are:
1. Hold tap on my shortcut on the (Google Pixel system default) Launcher, in my favorites bar
2. Tap "New Note"
3. Type a bit in the title and body, to test
4. Swipe up from the bottom to change to the Launcher, or to another app 
5. Repeat steps 1 and 2 expecting to see my note continued

It's empty 🙂

**Comment by alensiljak** [comment]
Interesting. Yes, that seems to be a wrong way. ;)

The difference in my setup is that I like to have a separate shortcut on the home screen. Orgzly has a shortcut for new note and I use that.
If you perform the steps above but use a separate shortcut instead of the menu item, then the unsaved note is displayed.
Mind you, this is not some insider knowledge but pure chance.

**Comment by amberin** [comment]
See also #271 - would be great if we can kill both with one stone.

**Comment by amberin** [comment]
The core issue discussed here was fixed by #678.  But I suppose a "drafts bin" could still be of use to some people. I'll leave it up to @Bujiraso to decide whether to close this issue or keep it open.

**Comment by Bujiraso** [comment]
Thanks @amberin , I was thinking this seemed fixed!

Let's close it. If someone wants a drafts bin in particular it may require further design, prototyping, planning, etc.

I'm happy to not lose my notes.
